### ISSUE TITLE
[Rule Tuning] Review Linux Rules for unquoted file paths that may cause FPs

### ISSUE THREAD

**Issue by w0rk3r** [elastic/detection-rules#2064]
## Link to rule

https://github.com/elastic/detection-rules/blob/main/rules/linux/privilege_escalation_ld_preload_shared_object_modif.toml
https://github.com/elastic/detection-rules/blob/main/rules/macos/persistence_periodic_tasks_file_mdofiy.toml
https://github.com/elastic/detection-rules/blob/main/rules/macos/privilege_escalation_root_crontab_filemod.toml

And potentially others.

## Description

A customer reported that the usage of `file.path` with unquoted paths is causing False Positives because of partial matches. This issue will be used to track the review of our ruleset for potential tunings. The customer is using auditbeat data, so we need to test with both endpoint and auditbeat.

**Comment by shashank-elastic** [comment]
In path for 8.5, this needs to be checked again

**Comment by w0rk3r** [comment]
After analyzing the issue with the customer, we found that this issue is likely caused by a custom/legacy mapping with the `file.path` as a text field. So after discussing, we decided to document the issue to help other customers that run through the same issue but not modify our detection rules.

Text fields, by default, use the [standard tokenizer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-tokenizer.html), which causes them to ignore most of the non-alphabet characters (hyphens, slashes, backslashes, etc.) on queries because it considers them delimiters and essentially splits the intended match into partial matches. Here is an [issue](https://github.com/elastic/integrations/issues/1776) that dealt with the same problem for our PowerShell integration.

The solution here is to update the mapping so the `file.path` field can be of the Keyword type, which **requires** the data to be reindexed.
